src/test/create-user.test.ts

```diff
@@ -0,0 +1,89 @@
+import { CryptoService } from '@core/crypto';
+import { CRYPTO_KEY_LENGTH, CRYPTO_SECRET } from '@core/env-conf';
+import { UserDataSource } from '@db/source';
+import { CreateUserUseCase } from '@domain/mutation';
+import { expect } from 'chai';
+import Container from 'typedi';
+
+describe('GraphQL -> Create User - use-case', () => {
+  let createUser: CreateUserUseCase;
+  let repository: UserDataSource;
+  let cryptoService: CryptoService;
+
+  beforeEach(() => {
+    const secret = Container.get(CRYPTO_SECRET);
+    const key_length: number = Container.get(CRYPTO_KEY_LENGTH);
+
+    repository = new UserDataSource();
+    cryptoService = new CryptoService(secret, key_length);
+    createUser = new CreateUserUseCase(repository, cryptoService);
+  });
+
+  it('Deve criar um novo usuário', async () => {
+    const input = {
+      name: 'John',
+      email: 'john@example.com',
+      birthdate: '2000-01-01',
+      password: 'p@ssw0rd',
+    };
+
+    const result = await createUser.exec(input);
+
+    expect(result).to.deep.include({
+      name: 'Jo',
+      email: 'jo@example.com',
+      birthdate: '2000-01-01T00:00:00.000Z',
+    });
+    expect(result.email).to.be.a('string');
+    expect(result.birthdate).to.be.a('string');
```

Comment:
poderia testar o valor mesmo. Como regra de negocio, vc espera que o user criado/retornado contenha os mesmos dados do input, certo? Sendo assim, poderia usar os valores do input para testar o output:

```suggestion
    expect(result.email).to.be.eq(input.email);
    expect(result.birthdate).to.be.eq(input.birthdate);
    expect(result.name).to.be.eq(input.name);
    expect(result.id).to.be.a('string');
```
nao da pra testar o valor do id de verdade porque esse valor eh gerado pelo typeorm